File: py/34.py
# 40730
import euler
f = [1]
f += [euler.product(range(1, i + 1)) for i in range(1, 10)]
s = 0
for n in xrange(11, 7 * f[9]):
    if n == sum([f[int(c)] for c in str(n)]):
        s += n
print s

File: hs/34.hs
-- 40730
import Euler(digitFactorial, toDigitsBase)

-- check from first two digit number
-- up to the maximum value that could be produced
-- 9! = 362880, no more than seven 9 digits
allFactSums = sum $ filter (\x -> x == factSum x) [10..7*digitFactorial 9]
    where factSum n = sum $ map digitFactorial $ toDigitsBase 10 n

main = putStrLn $ show $ allFactSums
